src/common/commandExecutor.ts

```diff
@@ -96,33 +94,32 @@ export class CommandExecutor {
             }
         });
 
-        Q.delay(300).then(() => deferred.resolve(result.spawnedProcess));
+        // TODO - PROMISE: We need to consume result.outcome here
+        Q.delay(300).done(() => deferred.resolve(result.spawnedProcess));
         return deferred.promise;
     }
 
     /**
      * Kills the React Native packager in a child process.
      */
-    public killReactPackager(packagerProcess: ChildProcess, outputChannel?: OutputChannel): void {
+    public killReactPackager(packagerProcess: ChildProcess, outputChannel?: OutputChannel): Q.Promise<void> {
         Log.logMessage("Stopping Packager", outputChannel);
 
         if (packagerProcess) {
             /* To reliably kill the child process on all versions of Windows,
              * please use taskkill to end the packager process */
             if (process.platform === "win32") {
-                new Node.ChildProcess().exec("taskkill /pid " + packagerProcess.pid + " /T /F").outcome.then(() => {
+                return new Node.ChildProcess().exec("taskkill /pid " + packagerProcess.pid + " /T /F").outcome.then(() => {
                     Log.logMessage("Packager stopped", outputChannel);
-                }, function() {
-                    Log.logError("Failed to exit the React Native packager", outputChannel);
                 });
             } else {
                 packagerProcess.kill();
                 Log.logMessage("Packager stopped", outputChannel);
+                return Q.resolve<void>(void 0);
```

Comment:
We need a return of a valid promise, because if not the code crashes. We can write that promise in different ways like:
return Q.resolve<void>(void 0);
return Q<void>(void 0);
etc...